--- freecraft-020630/setup.orig	Sat Dec 21 23:40:21 2002
+++ freecraft-020630/setup	Sun Dec 22 00:35:55 2002
@@ -443,7 +443,12 @@
   fi
   echo
   echo -n "Please enter selection: "
+  if [ -n "$WITH_VIDEO" ]; then
+	  VIDSEL="$WITH_VIDEO"
+	  echo $VIDSEL
+  else
   read $N VIDSEL
+  fi
   echo
 fi
 
@@ -551,7 +556,12 @@
 
 echo
 echo -n "Do you want to compile with sound? (y) "
+if [ -n "$WITH_SOUND" ]; then
+	ANSWER="$WITH_SOUND"
+	echo $ANSWER
+else
 read $N ANSWER
+fi
 echo
 case $ANSWER in
   [nN])
@@ -572,7 +582,12 @@
 	    echo
 	    echo "Assuming you have a thread-safe X11 (libc6 or glibc),"
 	    echo -n "do you want to compile with threaded sound? (n) "
+	    if [ -n "$WITH_THREADEDSOUND" ]; then
+		    ANSWER2="$WITH_THREADEDSOUND"
+		    echo $ANSWER2
+	    else
 	    read $N ANSWER2
+	    fi
 	    echo
 	    case $ANSWER2 in [yY])
 	      echo "THREAD		= -D_REENTRANT -DUSE_THREAD" >> $RULESFILE
@@ -589,7 +604,12 @@
 	LIBCHECK="-lFLAC -lm"; check_for_libs
 	if test "$?" = 0 ; then 
 	echo -n "Would you like FLAC support? (n) "
+	if [ -n "$WITH_FLAC" ]; then
+		ANSWER="$WITH_FLAC"
+		echo $ANSWER
+	else
 	read $N ANSWER
+	fi
 	echo
 	case $ANSWER in 
 	    [yY])
@@ -607,7 +627,12 @@
 	LIBCHECK="-lvorbisfile -lvorbis -logg"; check_for_libs
 	if test "$?" = 0 ; then 
 	echo -n "Would you like OGG support? (n) "
+	if [ -n "$WITH_OGG" ]; then
+		ANSWER="$WITH_OGG"
+		echo $ANSWER
+	else
 	read $N ANSWER
+	fi
 	echo
 	case $ANSWER in 
 	    [yY])
@@ -637,7 +662,12 @@
 	LIBCHECK="-lmad"; check_for_libs
 	if test "$?" = 0 ; then 
 	echo -n "Would you like MAD MP3 support? (n) "
+	if [ -n "$WITH_MAD" ]; then
+		ANSWER="$WITH_MAD"
+		echo $ANSWER
+	else
 	read $N ANSWER
+	fi
 	echo
 	case $ANSWER in 
 	    [yY])
@@ -657,7 +687,12 @@
 	echo -n "or [S]DL "
 	fi
 	echo -n "CD Audio support? (n) "
+	if [ -n "$WITH_CDA" ]; then
+		CDMode="$WITH_CDA"
+		echo $CDMode
+	else
 	read $N CDMode
+	fi
 	echo
 	case $CDMode in
 	    [sS])
@@ -741,7 +776,12 @@
 fi
 echo
 echo -n "Please enter selection ($COMPDEF): "
+if [ -n "$WITH_COMP" ]; then
+	COMPOPT="$WITH_COMP"
+	echo $COMPOPT
+else
 read $N COMPOPT
+fi
 echo
 
 if test "$COMPOPT x" = " x" ; then COMPOPT=$COMPDEF; fi
@@ -897,7 +937,12 @@
 
 echo
 echo -n "Are you using the WarCraft 2 Expansion CD? (n): "
+if [ -n "$WITH_WC2EXPCD" ]; then
+	YESORNO="$WITH_WC2EXPCD"
+	echo $YESORNO
+else
 read $N YESORNO
+fi
 echo
 if [ "$YESORNO" = "Y" -o "$YESORNO" = "y" ] ; then
 echo -n " -DHAVE_EXPANSION" >> $RULESFILE
@@ -965,7 +1010,12 @@
 echo "Done generating $RULESFILE."
 echo
 echo -n "Would you like to compile it now? (y) "
+if [ -n "$WITH_COMPILE" ]; then
+	ANSWER="$WITH_COMPILE"
+	echo $ANSWER
+else
 read $N ANSWER
+fi
 echo
 
 case $ANSWER in
