--- freecraft-020630/src/sound/sound_server.c.orig	Sun Jun 16 19:58:15 2002
+++ freecraft-020630/src/sound/sound_server.c	Sun Dec 22 00:44:57 2002
@@ -1083,7 +1083,7 @@
 	    audio_buf_info info;
 
 	    ioctl(SoundFildes, SNDCTL_DSP_GETOSPACE, &info);
-	    DebugLevel0("Free bytes %d\n", info.bytes);
+	    DebugLevel0("Free bytes %d\n" _C_ info.bytes);
 	}
 	FillChannels(free_channels, &discarded_request, &started_request);
 	how_many_playing += started_request;
@@ -1120,7 +1120,7 @@
 	    how_many_playing -= new_free_channels;
 	}
 	free_channels = MaxChannels - how_many_playing;
-	DebugLevel3("Channels: %d %d %d\n", free_channels, how_many_playing,
+	DebugLevel3("Channels: %d %d %d\n" _C_ free_channels _C_ how_many_playing _C_
 	    new_free_channels);
 	new_free_channels += discarded_request;
 	// decrement semaphore by the number of stopped channels
